### ISSUE TITLE
IndexError when using Ransac.fit()

### ISSUE THREAD

**Issue by mscheltienne** [autoreject/autoreject#224]
I ran into an issue with my data which I could reproduce using a dummy example with MNE sample data:

```python
from pathlib import Path

import mne
from autoreject import Ransac

sample_data_dir = Path(mne.datasets.sample.data_path())
fname = sample_data_dir / 'MEG' / 'sample' / 'sample_audvis_raw.fif'

raw = mne.io.read_raw_fif(fname)
raw.del_proj()
raw.pick(('eeg', 'stim'))

raw.set_eeg_reference(ref_channels='average', ch_type='eeg', projection=True)

events = mne.find_events(raw, stim_channel='STI 001')
epochs = mne.Epochs(
    raw, events, event_id=5, picks='eeg', tmin=-0.2, tmax=0.8,
    reject=None, verbose=False, detrend=None, proj=True,
    baseline=(None, 0), preload=True)

ransac = Ransac(verbose=False, picks='eeg', n_jobs=1)
ransac.fit(epochs)
```

Error: 

> IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices

Traceback:

> File "/var/folders/b1/kx2x00x95ls80zzkct4647sh0000gn/T/ipykernel_4007/3370373497.py", line 22, in <module>
>     ransac.fit(epochs)
> 
>   File "/Users/scheltie/Documents/pyvenv/neurotin/lib/python3.8/site-packages/autoreject/ransac.py", line 187, in fit
>     _check_data(epochs, picks=self.picks,
> 
>   File "/Users/scheltie/Documents/pyvenv/neurotin/lib/python3.8/site-packages/autoreject/utils.py", line 40, in _check_data
>     picked_info = mne.io.pick.pick_info(epochs.info, picks)
> 
>   File "<decorator-gen-8>", line 24, in pick_info
> 
>   File "/Users/scheltie/Documents/pyvenv/neurotin/lib/python3.8/site-packages/mne/io/pick.py", line 540, in pick_info
>     n_unique = len(np.unique(np.arange(len(info['ch_names']))[sel]))
> 
> IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices

I'm using the latest version of `MNE` (0.23) and `autoreject` (0.2.1).

**Comment by mscheltienne** [comment]
I feel like there is a `pick_to_idx` missing as the `sel` argument passed to `mne.io.pick.pick_info()` is not a list of id but the string `'eeg'`.
I think it should be placed in `autoreject.utils._handle_picks(info, picks)`, which currently looks like:
```
def _handle_picks(info, picks):
    """Pick the data channls or return picks."""
    if picks is None:
        out = mne.pick_types(
            info, meg=True, eeg=True, ref_meg=False, exclude='bads')
    else:
        out = picks
    return out
```
When `picks` is None, a selection (list of ID) is correctly returned. But when `picks` are provided, it directly returns the inputted string.

If someone can confirm this, I can open a PR for a fix.

**Comment by jasmainak** [comment]
yeah the `picks` in `autoreject` is ancient. Lots have changed in MNE since then. It used to be only list of int. If you can give a shot at fixing this with tests, that would be great!

**Comment by mscheltienne** [comment]
@jasmainak Ok so that is why. I'll give it a shot.

**Comment by jasmainak** [comment]
closed by #225
